google-home-community.groovy

```diff
@@ -328,12 +328,47 @@ def deviceTypePreferences(deviceType) {
                 )
             }
 
-            def useDevicePinCodes = false
-            deviceType?.traits.each { useDevicePinCodes = it.value?.useDevicePinCodes }
-            if ((deviceType?.secureCommands || deviceType?.pinCodes)) {
-                // device attribute is set to use app pincodes
-                if (useDevicePinCodes == false) {
-                    section("PIN Codes") {
+            if (deviceType?.secureCommands || deviceType?.pinCodes) {
+                section {
+                    input(
+                        name: "${devicePropertyName}.useDevicePinCodes",
+                        title: "Select to use device driver pincodes.  Deselect to use Google Home Community app pincodes.",
+                        type: "bool",
+                        defaultValue: "false",
+                        submitOnChange: true
+                    )
+                }
+
+                if (deviceType?.useDevicePinCodes == true) {
+                    // device attribute is set to use device driver pincodes
+                    section("PIN Codes (Device Driver)") {
+                        input(
+                            name: "${devicePropertyName}.pinCodeAttribute",
+                            title: "Device pin code attribute",
+                            type: "text",
+                            defaultValue: "lockCodes",
+                            required: true
+                        )
+
+                        input(
+                            name: "${devicePropertyName}.pinCodeValue",
+                            title: "Device pin code value",
+                            type: "text",
+                            defaultValue: "code",
+                            required: true
+                        )
+                        input(
+                            name: "${devicePropertyName}.returnUserIndexToDevice",
+                            title: "Select to return the user index with the device command on a pincode match. " +
+                                    "Not all device drivers support this operation.",
+                            type: "bool",
+                            defaultValue: false,
+                            submitOnChange: true
+                        )
```

Comment:
Does this setting make sense at the device type level?  Only Lock/Unlock and Arm/Disarm support it.

Reply:
I've moved it back to the trait level for those two traits.